saleor/graphql/checkout/tests/test_checkout_lines.py

```diff
@@ -512,6 +512,26 @@ def test_checkout_lines_update_variant_quantity_over_avability_stock(
     assert data["checkout"]["lines"][0]["quantity"] == variables["lines"][0]["quantity"]
 
 
+def test_checkout_lines_delete_with_by_zero_quantity_when_variant_out_of_stock(
+    user_api_client, checkout_with_item
+):
+    checkout = checkout_with_item
+    line = checkout.lines.first()
+    variant_id = graphene.Node.to_global_id("ProductVariant", line.variant.pk)
+    line.quantity = 0
+    line.save()
```

Comment:
You can use `update_fields` in `save`:
```suggestion
    line.save(update_fields=["quantity"])
```